File: src/main/java/ast/service/FileService.java
package ast.service;

import ast.repository.NonPersistentRepository;
import ast.repository.model.File;
import com.google.inject.Inject;

public class FileService {

    @Inject
    private NonPersistentRepository<File> fileRepository;

    public void addFile(File file) {
        fileRepository.addObject(file);
    }

}

File: src/main/java/org/vladg/ast/service/FileService.kt
package org.vladg.ast.service

import com.google.inject.Inject
import org.vladg.ast.repository.NonPersistentRepository
import org.vladg.ast.repository.model.File

class FileService {

    @Inject
    private lateinit var fileRepository: NonPersistentRepository<File>

    fun addFile(file: File) {
        fileRepository.addObject(file)
    }

}